packages/core/src/index.ts

```diff
@@ -234,7 +234,7 @@ export class Server<TContext extends InitialContext, TRootValue> {
     if (typeof options.graphiql === 'object' || options.graphiql === false) {
       this.graphiql = options.graphiql
     } else {
-      this.graphiql = introspectionEnabled ? {} : false
+      this.graphiql = {}
```

Comment:
graphiql should be disabled by default if introspection is disabled.

Reply:
After discussion with @dotansimha we decided those should be separate (Graphiql has it's own flag)

reason: `introspectionEnabled` (which is replaced with `introspectionDisabled`) can be defined by dynamic condition function, and Graphiql can't be configured dynamically